src/main/java/frc/robot/commands/JoystickTarget.java

```diff
@@ -0,0 +1,59 @@
+package frc.robot.commands;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.GenericHID;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.Constants.OIConstants;
+import frc.robot.Constants.SwerveModuleConstants;
+import frc.robot.Constants.VisionConstants;
+import frc.robot.subsystems.DriveSubsystem;
+import frc.robot.subsystems.Limelight;
+
+public class JoystickTarget extends Command {
+  private DriveSubsystem m_drive;
+  private Limelight m_limelight;
+  private GenericHID m_driverController;
+  private ProfiledPIDController thetaController = new ProfiledPIDController(
+      SwerveModuleConstants.kPTurningController,
+      SwerveModuleConstants.kITurningController,
+      SwerveModuleConstants.kDTurningController,
+      new TrapezoidProfile.Constraints(
+        SwerveModuleConstants.kMaxAngularSpeedRadiansPerSecond,
+        SwerveModuleConstants.kMaxAngularAccelerationRadiansPerSecondSquared));
+  
+  public JoystickTarget(DriveSubsystem drive, Limelight limelight, GenericHID driverController) {
+    m_drive = drive;
+    m_limelight = limelight;
+    m_driverController = driverController;
+    addRequirements(drive, limelight);
+  }
+
+  @Override
+  public void initialize() {
+    thetaController.reset(0);
+    thetaController.setTolerance(VisionConstants.kTargetingTolerance);
+  }
+
+  @Override
+  public void execute() {
+    double x = m_limelight.getX();
+    double xVelocity = thetaController.calculate(Units.degreesToRadians(x));
+    double reverseFactor = -1.0;
+    m_drive.drive(
+      reverseFactor * joystickTransform(m_driverController.getRawAxis(OIConstants.kLeftJoyYAxis)) * OIConstants.kMaxMetersPerSec,
+      reverseFactor * joystickTransform(m_driverController.getRawAxis(OIConstants.kLeftJoyXAxis)) * OIConstants.kMaxMetersPerSec,
+      -joystickTransform(m_driverController.getRawAxis(OIConstants.kRightJoyXAxis)) + xVelocity,
```

Comment:
Do I understand correctly that it's still possible to manually rotate the robot, but that if a note is visible the robot will "fight" with the manual input? I wonder how well the robot would behave if we had it start ignoring the driver's rotation input as soon as a note is visible.

Reply:
Yeah, I removed this in a later commit.